org.adempiere.base/src/org/compiere/model/MPayment.java

```diff
@@ -1937,12 +1937,6 @@ public String prepareIt()
 				int length = invoices.length;
 				if (length > 0)		//	get last invoice
 					setC_Invoice_ID (invoices[length-1].getC_Invoice_ID());
-				//
-				if (getC_Invoice_ID() == 0)
-				{
-					m_processMsg = "@NotFound@ @C_Invoice_ID@";
-					return DocAction.STATUS_Invalid;
-				}

```

Comment:
why the need to remove this validation ?

Reply:
I consulted with Igor, he said that switching off the invoice generating (with IsAutoGenerateInvoice) would cause to return STATUS_Invalid here on the payment, and would roll back everything

Reply:
If that's the case, the validation should only be removed for order with doctype.isautogenerateinvoice == false

Reply:
Returned the validation, extended with the IsAutoGenerateInvoice check